docopt/docopt.c

```diff
@@ -4,9 +4,40 @@
 #include <stdio.h>
 #include <ctype.h>
 #include <assert.h>
-#ifdef __linux__
+
```

Comment:
Why all of this is needed? Linux/limits.h? I don't remember I put something system dependent. Kernel version check? This is simple generator, plain c source, no system code.

Reply:
It is needed for `PATH_MAX` as used https://github.com/offscale/docopt.c-yacc-lex/blob/b8af845/docopt/docopt.c#L1127-L1128

See also: https://pubs.opengroup.org/onlinepubs/009695399/basedefs/limits.h.html https://github.com/torvalds/linux/blob/master/include/linux/limits.h

Reply:
Great, but why not to include limits.h under linux without kernel version checks?

Reply:
You mean it might be overkill to include support for Linux kernel versions less than or equal to 2.6.22? - Yeah maybe 2007 is too old a version to support… http://lkml.iu.edu/hypermail/linux/kernel/0707.1/0195.html

Reply:
No, I'm telling that just include limits.h since PATH_MAX is defined there, without checking the kernel version. We don't use any naked syscalls or kernel features.

Reply:
Oh wait it was there in kernel v2 https://elixir.bootlin.com/linux/2.0/source/include/linux/limits.h#L14; my bad

Reply:
Samuel, please understand me right, I don't want to spoil the code with ugly architectural includes or defines. What operating system are you using every day? Do you use all the set of operating systems you listed? If you care about windows - please send me patches for windows. If your choice is arm board with Debian and there are some compilation errors - I'm open for patches. But let's not make support of all available oses of all times out of the simple code. Linux v0. 97? What? Who needs that?

Let's try to fix real problems for real requirements.

Thanks.